crates/daphne/src/roles/leader/mod.rs

```diff
@@ -223,6 +223,47 @@ pub async fn handle_upload_req<A: DapLeader>(
         .into());
     }
 
+    match (
+        &report.report_metadata.public_extensions,
+        task_config.version,
+    ) {
+        (Some(extensions), DapVersion::Latest) => {
+            let mut unknown_extensions = Vec::<u16>::new();
+            if crate::protocol::no_duplicates(extensions.iter()).is_err() {
```

Comment:
And `use crate::protocol;`. Also, we should rename `no_duplicates` to something more distinctive.

```suggestion
            if protocol::no_duplicates(extensions.iter()).is_err() {
```